### PR TITLE
Missing Data plot fixes

### PR DESCRIPTION
**Pull request by ngupta23**
## Related Issuse or bug

- Partially Implements #2364
- Also fixed a bug in the retrieval of test values

#### Describe the changes you've made

- Added ability for `ts` and `train_test_split` plots to accept different plot data types - "original", "imputed", "transformed"
- Also fixed a bug in the retrieval of test values - these should be computed by passing the entire dataset through the pipeline and then retrieving the transformed test set values instead of just the test set through the pipeline (since we can use all past data but no future data)dataset values

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Unit tests have been added for plotting as well as pipeline

## Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

File: pycaret/internal/plots/utils/time_series.py [hunk 1]
```diff
@@ -24,6 +25,46 @@
 PlotReturnType = Tuple[Optional[go.Figure], Optional[Dict[str, Any]]]
 
 
+#### Data Types allowed for each plot type ----
+# First one in the list is the default (if requested is None)
+ALLOWED_PLOT_DATA_TYPES = {
```
Comment (Yard1) [inline pycaret/internal/plots/utils/time_series.py:30]: Maybe we can use enums here?

**Review by Yard1 — COMMENTED** [review]
Logic looks good, but I think we whouls be using enums for data types and split types instead of regular strings (ofc we should allow the user to pass in a string, but we should use enums in the code). Let me know what you think

File: pycaret/internal/plots/utils/time_series.py (hunk 1 above)
Comment (ngupta23) [reply pycaret/internal/plots/utils/time_series.py:30]: ok, makes sense. Maybe I will add that as a final refactor to make it easier since I am in the middle of more changes. Hope that is fine.

Comment (ngupta23) [reply pycaret/internal/plots/utils/time_series.py:30]: I opened another issue for this for tracking. Does this only apply to ALLOWED_PLOT_DATA_TYPES? 
Also, are you ok merging this if we do this refactor later?
